builtin/credential/ldap/backend.go

```diff
@@ -118,9 +118,25 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 	if cfg.BindDN != "" && cfg.BindPassword != "" {
 		if err := c.Bind(cfg.BindDN, cfg.BindPassword); err != nil {
 			if b.Logger().IsDebug() {
-				b.Logger().Debug("error while attempting to re-bind with the BindDN User", "error", err)
+				b.Logger().Debug("error while attempting to re-bind with the BindDN User; trying with new connection", "error", err)
+			}
+			// Create a new client object to work around an issue with
+			// go-ldap and DUO's LDAP proxy. It is not known whether this is
+			// a bug with go-ldap or with DUO's proxy.
+			c, err = ldapClient.DialLDAP(cfg)
+			if err != nil {
+				return nil, logical.ErrorResponse(err.Error()), nil, nil
+			}
+			if c == nil {
+				return nil, logical.ErrorResponse("invalid connection returned from LDAP dial"), nil, nil
```

Comment:
This should be a `return nil, nil, nil, errors.New("invalid connection returned from LDAP dial")`